app/models/order_item.rb

```diff
@@ -52,4 +64,11 @@ def service_parameters_raw=(val)
   def parameters_contain_sanitized_values?(val)
     val.to_s.include?("\"#{Catalog::OrderItemSanitizedParameters::MASKED_VALUE}\"")
   end
+
+  def mark_item(msg = nil, level: "info", **opts)
+    update!(**opts)
+    update_message(level, msg) if msg
+
+    Catalog::OrderStateTransition.new(order_id).process
```

Comment:
Now that this is moved into lib should we just pass the `order` instead of the `order_id`? `OrderStateTransition` does a `find` with the `order_id`, but since it's already available that seems like extra overhead that we could avoid.

Reply:
Updated!